packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -123,11 +124,52 @@ class CupertinoCheckbox extends StatefulWidget {
 
   /// The color to use when this checkbox is checked.
   ///
+  /// If [fillColor] returns a non-null color in the [WidgetState.selected]
+  /// state, it will be used instead of this color.
+  ///
   /// Defaults to [CupertinoColors.activeBlue].
   final Color? activeColor;
 
+  /// {@template flutter.cupertino.CupertinoCheckbox.fillColor}
+  /// The color used to fill this checkbox.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [fillColor] based on the current [WidgetState]
+  /// of the [CupertinoCheckbox], providing a different [Color] when it is
+  /// [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// CupertinoCheckbox(
+  ///   value: true,
+  ///   onChanged: (_){},
+  ///   fillColor: WidgetStateProperty.resolveWith<Color>((Set<WidgetState> states) {
+  ///     if (states.contains(WidgetState.disabled)) {
+  ///       return Colors.orange.withOpacity(.32);
+  ///     }
+  ///     return Colors.orange;
+  ///   })
+  /// )
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state. If that is also null, then [CupertinoColors.inactiveGray] is used in
+  /// the unselected state, and [CupertinoColors.activeBlue] is used in the
+  /// selected state.
+  final WidgetStateProperty<Color?>? fillColor;
```

Comment:
Also, can someone explain to me why this is `WidgetStateProperty<Color?>?` instead of `WidgetStateProperty<Color>?`? I think @victorsanni explained to me once but either the discussion was interrupted or I didn't fully understand at that time. My idea is that either the developer provides `fillColor` to specify all colors or the developer doesn't use `fillColor` at all. There is little reason for them to use `fillColor` for some of the colors and `activeColor` for the rest.

Reply:
The primary reason is to match Material `Checkbox`s `fillColor`, so that later adaptive work is easier. 

Also, there might be situations where the user needs `fillColor` but doesn't use it for selected/unselected states? For example, setting the color only in `WidgetState.disabled`, without being forced to also set the color in every other state (and just defer to the defaults).

Reply:
I wonder if @johnsonmh , who added the Material `Checkbox.fillColor` can shed some light on this design. 

That said, while I still think a `<Color>` is a simpler API, I agree that using the `<Color?>` to match the material API reduces the cognitive load. We might as well follow the convention and discuss changes as a separate topic.